Compiler flags are passed as is, not matter the compiler

At the moment, the compilation flags are passed as `gcc` expects them. However, `clang` does not expect some of them, or has its own version.

Clang version:
```
Debian clang version 19.1.4 (1~deb12u1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-19/bin
```
CMake version: `cmake version 3.25.1`

```sh
42sh$ CXX="$(which clang++-19)" cmake -B build
-- The CXX compiler identification is Clang 19.1.4
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-19 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/g704120/cxx_tuncfest/samples/simple/build
42sh$ make -C build/
make: Entering directory '/home/g704120/cxx_tuncfest/samples/simple/build'
make[1]: Entering directory '/home/g704120/cxx_tuncfest/samples/simple/build'
make[2]: Entering directory '/home/g704120/cxx_tuncfest/samples/simple/build'
make[2]: Leaving directory '/home/g704120/cxx_tuncfest/samples/simple/build'
make[2]: Entering directory '/home/g704120/cxx_tuncfest/samples/simple/build'
[ 50%] Building CXX object CMakeFiles/main.dir/main.cc.o
error: unknown warning option '-Wlogical-op'; did you mean '-Wlong-long'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wformat-overflow=2'; did you mean '-Wformat-overflow'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wformat-truncation=2'; did you mean '-Wformat-truncation'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wshift-overflow=2'; did you mean '-Wshift-overflow'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wuseless-cast' [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wsuggest-attribute=const' [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wsuggest-attribute=noreturn' [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wsuggest-attribute=format'; did you mean '-Wproperty-attribute-mismatch'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wsuggest-attribute=cold'; did you mean '-WNSObject-attribute'? [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Warith-conversion' [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wduplicated-cond' [-Werror,-Wunknown-warning-option]
error: unknown warning option '-Wunsafe-loop-optimizations' [-Werror,-Wunknown-warning-option]
make[2]: *** [CMakeFiles/main.dir/build.make:76: CMakeFiles/main.dir/main.cc.o] Error 1
make[2]: Leaving directory '/home/g704120/cxx_tuncfest/samples/simple/build'
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/main.dir/all] Error 2
make[1]: Leaving directory '/home/g704120/cxx_tuncfest/samples/simple/build'
make: *** [Makefile:91: all] Error 2
make: Leaving directory '/home/g704120/cxx_tuncfest/samples/simple/build'
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,22 +12,27 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
         message(FATAL_ERROR "Clang version 16.0 or higher is required for C++23 support")
     endif()
 else()
-    message(FATAL_ERROR "Unsupported compiler. Please use GCC 13.0 or higher, or Clang 16.0 or higher.")
+    message(FATAL_ERROR "Unsupported compiler. Please use GCC 13.0+ or Clang 16.0+.")
 endif()
 
 set(CMAKE_CXX_STANDARD 23)
-set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
-    -fdiagnostics-color=always -Werror -Wall -Wextra -pedantic -Wfloat-equal -Wundef -Wshadow -Wswitch-default -Wlogical-op -Wdouble-promotion -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wno-format-zero-length -Wmissing-include-dirs -Wshift-negative-value -Wshift-overflow=2 -Wuseless-cast -Wstring-compare -Wsuggest-attribute=const -Wsuggest-attribute=noreturn -Wmissing-noreturn -Wsuggest-attribute=format -Wsuggest-attribute=cold -Warith-conversion -Wduplicated-cond -Warray-bounds -Wunsafe-loop-optimizations -Wdangling-else -Wenum-conversion"
-)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-command-line-argument")
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+set(WARN_FLAGS "-fdiagnostics-color=always -Wall -Wextra -Wpedantic -Werror")
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(WARN_FLAGS "${WARN_FLAGS} -Wfloat-equal -Wundef -Wshadow -Wswitch-default -Wlogical-op -Wdouble-promotion -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wno-format-zero-length -Wmissing-include-dirs -Wshift-negative-value -Wshift-overflow=2 -Wuseless-cast -Wstring-compare -Wsuggest-attribute=const -Wsuggest-attribute=noreturn -Wmissing-noreturn -Wsuggest-attribute=format -Wsuggest-attribute=cold -Warith-conversion -Wduplicated-cond -Warray-bounds -Wunsafe-loop-optimizations -Wdangling-else -Wenum-conversion")
 endif()
 
-add_library(tuncfest INTERFACE)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    set(WARN_FLAGS "${WARN_FLAGS} -Wfloat-equal -Wundef -Wshadow -Wswitch-enum -Wlogical-op-parentheses -Wdouble-promotion -Wformat=2 -Wmissing-noreturn -Wcast-align -Wcast-qual -Wstring-compare -Wcovered-switch-default -Wunreachable-code -Wcomma -Wextra-semi -Wnewline-eof -Wno-unused-command-line-argument")
+endif()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARN_FLAGS}")
+
+add_library(tuncfest INTERFACE)
 target_include_directories(tuncfest
     INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/tuncfest>

--- a/samples/heavy/CMakeLists.txt
+++ b/samples/heavy/CMakeLists.txt
@@ -1,11 +1,35 @@
 cmake_minimum_required(VERSION 3.10)
-project(simple_example VERSION 0.1.0 LANGUAGES CXX)
+project(heavy_example VERSION 0.1.0 LANGUAGES CXX)
+
+# `static operator()`
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "13.0")
+        message(FATAL_ERROR "GCC version 13.0 or higher is required for C++23 support")
+    endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "16.0")
+        message(FATAL_ERROR "Clang version 16.0 or higher is required for C++23 support")
+    endif()
+else()
+    message(FATAL_ERROR "Unsupported compiler. Please use GCC 13.0+ or Clang 16.0+.")
+endif()
 
 set(CMAKE_CXX_STANDARD 23)
-set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always -Werror -Wall -Wextra -pedantic -Wfloat-equal -Wundef -Wshadow -Wswitch-default -Wlogical-op -Wdouble-promotion -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wno-format-zero-length -Wmissing-include-dirs -Wshift-negative-value -Wshift-overflow=2 -Wuseless-cast -Wstring-compare -Wsuggest-attribute=const -Wsuggest-attribute=noreturn -Wmissing-noreturn -Wsuggest-attribute=format -Wsuggest-attribute=cold -Warith-conversion -Wduplicated-cond -Warray-bounds -Wunsafe-loop-optimizations -Wdangling-else -Wenum-conversion"
-)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+set(WARN_FLAGS "-fdiagnostics-color=always -Wall -Wextra -Wpedantic -Werror")
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(WARN_FLAGS "${WARN_FLAGS} -Wfloat-equal -Wundef -Wshadow -Wswitch-default -Wlogical-op -Wdouble-promotion -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wno-format-zero-length -Wmissing-include-dirs -Wshift-negative-value -Wshift-overflow=2 -Wuseless-cast -Wstring-compare -Wsuggest-attribute=const -Wsuggest-attribute=noreturn -Wmissing-noreturn -Wsuggest-attribute=format -Wsuggest-attribute=cold -Warith-conversion -Wduplicated-cond -Warray-bounds -Wunsafe-loop-optimizations -Wdangling-else -Wenum-conversion")
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    set(WARN_FLAGS "${WARN_FLAGS} -Wfloat-equal -Wundef -Wshadow -Wswitch-enum -Wlogical-op-parentheses -Wdouble-promotion -Wformat=2 -Wmissing-noreturn -Wcast-align -Wcast-qual -Wstring-compare -Wcovered-switch-default -Wunreachable-code -Wcomma -Wextra-semi -Wnewline-eof -Wno-unused-command-line-argument")
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARN_FLAGS}")
 
 add_executable(myheavycat myheavycat.cc)
 

--- a/samples/simple/CMakeLists.txt
+++ b/samples/simple/CMakeLists.txt
@@ -6,7 +6,7 @@ set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 
 # Works with every compilation warning I have tested
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always -Werror -Wall -Wextra -pedantic -Wfloat-equal -Wundef -Wshadow -Wswitch-default -Wlogical-op -Wdouble-promotion -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wno-format-zero-length -Wmissing-include-dirs -Wshift-negative-value -Wshift-overflow=2 -Wuseless-cast -Wstring-compare -Wsuggest-attribute=const -Wsuggest-attribute=noreturn -Wmissing-noreturn -Wsuggest-attribute=format -Wsuggest-attribute=cold -Warith-conversion -Wduplicated-cond -Warray-bounds -Wunsafe-loop-optimizations -Wdangling-else -Wenum-conversion"
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always -Werror -Wall -Wextra -pedantic -Wfloat-equal -Wundef -Wshadow -Wswitch-default -Wdouble-promotion -Wformat=2 -Wformat-signedness -Wno-format-zero-length -Wmissing-include-dirs -Wshift-negative-value -Wstring-compare -Wmissing-noreturn -Warray-bounds -Wdangling-else -Wenum-conversion"
 )
 
 # Getting the tuncfest automagically

--- a/samples/simple/README.md
+++ b/samples/simple/README.md
@@ -34,12 +34,12 @@ int main(void)
 Then we define a tests Builder. Let's examine one:
 
 ```cpp
-constexpr auto greatTestBuilder = testBuilder("GreatTest")
+constexpr auto greatTestBuilder = TestBuilder<>()
+                                      .with_name<"GreatTest">()
                                       .with_stdinput<"42\n">()
                                       .with_expected_stdout<"42\n">();
 ```
 
-First, the testBuilder macro takes a string. That will be the Test's name.
 TestObject instances have a compile time fluent interface (not exactly since the
 final Test will be another type entirely but that's beyond the point) which let
 you change the final test's parameters, such as the command line arguments, the
@@ -49,10 +49,6 @@ expected stdout of the test, the exit code, etc.
 
 Same as before, the builders can be registered and created in the main function,
 and with other qualifiers, but they need to be realizable in compile time.
-
-You **should** keep the auto and not use the exact type of this, but
-in case you ever need it, it would be an instance of something like
-`TestBuilder<MakeTagImpl<sv("FirstTest")>::type, /* other parameters */>`.
 
 ### Registration
 
@@ -77,7 +73,7 @@ Thus, the REGISTER_TEST macro is simply:
 #define REGISTER_TEST(NAME, BUILDER) using NAME = decltype(BUILDER)::Result
 ```
 
-That is why a builder can be instantiate many different tests in between
+That is why a builder can instantiate many different tests in between
 modifications. Each time you modify a Builder, it really creates a new builder
 type with a different nested Result struct; and if every parameter is the same,
 it automagically fall back to the already created type with the same template

